src/librustdoc/html/render/print_item.rs

```diff
@@ -1571,6 +1573,35 @@ fn document_type_layout(w: &mut Buffer, cx: &Context<'_>, ty_def_id: DefId) {
                     pl = if bytes == 1 { "" } else { "s" },
                 );
             }
+            if let Variants::Multiple { variants, .. } = &ty_layout.layout.variants {
+                if !variants.is_empty() {
+                    w.write_str(
+                        "<p>\
+                            <strong>Size for each variant:</strong>\
+                            <ul>",
+                    );
+                    for (index, layout) in variants.iter_enumerated() {
+                        let ident = if let Adt(adt, _) = ty_layout.ty.kind() {
+                            adt.variants[index].ident
+                        } else {
+                            bug!("not an adt")
+                        };
+
+                        if layout.abi.is_unsized() {
+                            writeln!(w, "<li><code>{name}</code> (unsized)</li>", name = ident);
+                        } else {
+                            let bytes = layout.size.bytes();
+                            writeln!(
+                                w,
+                                "<li><code>{name}</code>: {size} byte{pl}</li>",
+                                name = ident,
+                                size = bytes,
+                                pl = if bytes == 1 { "" } else { "s" },
+                            );
+                        }
+                    }
```

Comment:
nit: you are missing the closing brackets for the `ul` and `p` tags
```suggestion
                    }
                    w.write_str("</ul></p>");
```